optuna/study/_optimize.py

```diff
@@ -219,48 +220,35 @@ def _run_trial(
         state = TrialState.FAIL
         func_err = e
         func_err_fail_exc_info = sys.exc_info()
-    else:
-        # TODO(hvy): Avoid checking the values both here and inside `Study.tell`.
-        values, values_conversion_failure_message = _check_and_convert_to_values(
-            len(study.directions), value_or_values, trial.number
-        )
-        if values_conversion_failure_message is not None:
-            state = TrialState.FAIL
-        else:
-            state = TrialState.COMPLETE
 
     if study._storage.is_heartbeat_enabled():
         assert stop_event is not None
         assert thread is not None
         stop_event.set()
         thread.join()
 
-    # `Study.tell` may raise during trial post-processing.
```

Comment:
Is changing this behavior (removing `finally` which ensures that results are always logged) also an intended part of this PR? Note that post-processing here refers to the after-trial sampler callback.

Reply:
Thank you for pointing out, this try-catch must not be removed.
03d80ea reverts the change, and 91d9424 fetches a frozen trial from the storage to avoid the unbounded error.

Reply:
Not a strong opinion but would the code become cleaner/no need for descriptive comment if we do this trial fetching inside the except clause/indent?